tests/lib/location/src/location_test.c

```diff
@@ -246,6 +246,40 @@ void test_location_method_str(void)
 	TEST_ASSERT_EQUAL_STRING("Unknown", method);
 }
 
+static bool expect_nrf_modem_at_scanf;
```

Comment:
I guess you'd need to set this to `false` somewhere (`tearDown`?) or otherwise we just expect this always after the first test using it sets it to `true`.

And if you want to check that this is not called when not expected, you'd need to do something.

Maybe something similar to location callback in `tearDown()` and initialization for those parameters in `setUp()`:
`	TEST_ASSERT_EQUAL(location_callback_called_expected, location_callback_called_occurred);`

This aspect is handled in PR #8285.

Reply:
Does this solution check if `nrf_modem_at_scanf` is called multiple times?

Reply:
> I guess you'd need to set this to false somewhere (tearDown?) or otherwise we just expect this always after the first test using it sets it to true.

It is set to `false` in the mock we made. But it is better to assert that it is always `false` in `tearDown()` function. That way we can test that it is called when it is expected. I will add this.

>And if you want to check that this is not called when not expected, you'd need to do something.

That is done already here -> https://github.com/nrfconnect/sdk-nrf/pull/8723/files#diff-e8a49aea1afea4afb8a674b5bbe255191601589c4bf48db44916284aeeb80d6bR257 

> Maybe something similar to location callback in tearDown() and initialization for those parameters in setUp():
TEST_ASSERT_EQUAL(location_callback_called_expected, location_callback_called_occurred);

I will do `TEST_ASSERT_FALSE(expect_nrf_modem_at_scanf);` (as mentioned above) That should fix this.

> Does this solution check if nrf_modem_at_scanf is called multiple times?

No. And this test suite does not need it afaik. It can be added later if necessary

Reply:
Seems everything is handled.

> > Does this solution check if nrf_modem_at_scanf is called multiple times?
> 
> No. And this test suite does not need it afaik. It can be added later if necessary

I meant a check that the function is not called multiple times when we expect just one call but `TEST_ASSERT_TRUE(expect_nrf_modem_at_scanf);` actually handles that part because `expect_nrf_modem_at_scanf` is set to `false` after the first call.

Reply:
yes. :)